### ISSUE TITLE
Terraform fails when update with cluster replacement and k8s ressources in usage

### ISSUE THREAD

**Issue by grieshaber** [terraform-google-modules/terraform-google-kubernetes-engine#1234]
### TL;DR

When using in-cluster ressources (e.g. kube-dns configmap through stub-domains parameter), running an update, that forces the replacement of the cluster and using the provided kubernetes-provider configuration (https://github.com/terraform-google-modules/terraform-google-kubernetes-engine#usage) the terraform steps (plan / apply) will fail:
```
│ Error: Unauthorized
│ 
│   with module.gke.module.gke.kubernetes_config_map.kube-dns[0],
│   on .terraform/modules/gke.gke/modules/private-cluster/dns.tf line 45, in resource "kubernetes_config_map" "kube-dns":
│   45: resource "kubernetes_config_map" "kube-dns" {
```
In our example, the kubernetes provider cannot retrieve a valid token and / or cluster endpoint and tries to connect to the local kubernetes cluster hosting the CI. Therefore the `Unauthorized`. On local machines, it will fails with: 

```
Error: Get "http://localhost/api/v1/namespaces/kube-system/configmaps/kube-dns": dial tcp [::1]:80: connect: connection refused
│
│   with module.gke.module.gke.kubernetes_config_map.kube-dns[0],
│   on .terraform/modules/gke.gke/modules/private-cluster/dns.tf line 45, in resource "kubernetes_config_map" "kube-dns":
│   45: resource "kubernetes_config_map" "kube-dns" {
```

We guess, the k8s provider cannot receive valid cluster information, if the cluster is tagged for replacement.

### Expected behavior

The update / upgrade will succeed without an error.

### Observed behavior

The update / upgrade is not able to connect to the kubernetes cluster.

### Terraform Configuration

```hcl
data "google_client_config" "default" {}

provider "kubernetes" {
  host                   = "https://${module.gke.cluster_endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke.cluster_ca_crt)
}

module "gke" {
  source     = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
  version    = "20.0.0"
  project_id = var.project_id
  ... 
}
```

### Terraform Version

```sh
✗  terraform version
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.18.0
+ provider registry.terraform.io/hashicorp/google-beta v4.3.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.8.0
```

**Comment by bharathkkb** [comment]
@grieshaber Thanks for the report. Does the cluster currently exit and`module.gke.cluster_endpoint` give a valid endpoint?

**Comment by grieshaber** [comment]
Hey @bharathkkb, yes it is alive and returns a valid (private) endpoint:
```
➜  terraform console
> nonsensitive(module.gke.cluster_endpoint)
"10.17.190.18"
```

**Comment by grieshaber** [comment]
Im currently testing the setup described here: https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/using_gke_with_terraform#using-the-kubernetes-and-helm-providers

I'll add the results afterwards.

**Comment by grieshaber** [comment]
Unfortunately, this does not work for installation from scratch, when there is no exisiting cluster.
